CMakeLists.txt

```diff
@@ -15,6 +15,10 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+if(BUILD_SHARED_LIBS OR CMAKE_POSITION_INDEPENDENT_CODE)
+  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Comment:
Why do you want to set it again?

Reply:
This needs to be set in order to prevent build errors in the submodules if calling cmake with conflicting options (eg:
-DBUILD_SHARED_LIBS=1 -DCMAKE_POSITION_INDEPENDENT_CODE=0). This way if BUILD_SHARED_LIBS is set, it will always build the library and submodules with PIC enabled.